---
# -------------------- localization ---------------------

ekf_odom_filter:
  ros__parameters:
    # Topics
    odom_topic: "/base_controller/odom"  # Input odometry topic
    imu_topic: "/imu"                    # Input IMU topic
    odom_filtered_topic: "/odom_filter"  # Output filtered odometry topic

    # EKF Parameters
    # initial_state: [0.0, 0.0, 0.0]       # Initial state [x, y, theta]
    process_noise: [0.1, 0.1]              # Process noise covariance (Linear, Angular)
    odom_position_noise: 0.1               # Odometry position noise
    imu_yaw_noise: 0.01                    # IMU yaw noise

    # Filter parameters
    alpha: 0.5                           # Low-pass filter alpha for velocities
    update_frequency: 5.0                # EKF update frequency in Hz

map_publisher:
  ros__parameters:
    image: /absolute/path/to/map.pgm
    resolution: 0.05
    origin: [-10.0, -10.0, 0.0]
    occupied_thresh: 0.65
    free_thresh: 0.2
    negate: 0
    frame_id: "map"

custom_amcl_node:
  ros__parameters:
    # Map parameters
    map_topic: "/map"

    # Laser scan parameters
    laser_scan_topic: "/scan"

    # Odometry topic
    odom_topic: "/odom_filter"

    # Localization settings
    num_particles: 200         # Number of particles in the particle filter
    map_resolution: 0.05       # Map resolution in meters per cell
    
    motion_noise: [0.05, 0.05]
    sensor_noise: 0.2
    resampling_threshold: 0.5
    particle_spread: [0.5, 0.5]
    scan_matching_weight: [1.0, 0.5]
    adaptive_particles: true


# ------------------ mapping --------------------
scan_to_map_node:
  ros__parameters:
    # LaserScan Parameters
    scan_topic: "/scan"              # Topic to subscribe to for LaserScan data
    map_topic: "/map"                # Topic to publish the OccupancyGrid map
    range_min: 0.1                   # Minimum valid range for laser scan
    range_max: 12.0                  # Maximum valid range for laser scan

    # Map Parameters
    map_resolution: 0.05             # Resolution of the map (meters per cell)
    map_width: 200                   # Width of the map in cells
    map_height: 200                  # Height of the map in cells
    map_origin_x: -5.0               # X-coordinate of the map's origin
    map_origin_y: -5.0               # Y-coordinate of the map's origin

    # Frame Parameters
    map_frame: "map"                 # Frame ID for the OccupancyGrid map
    robot_frame: "base_link"         # Frame ID for the robot
